lib/cext/ruby/ruby.h

```diff
@@ -527,16 +527,18 @@ int rb_type(VALUE value);
 bool RB_TYPE_P(VALUE value, int type);
 
 
-/* RB_GC_GUARD_PTR() is an intermediate macro, and has no effect by
- * itself.  don't use it directly */
 #ifdef __GNUC__
-#define RB_GC_GUARD_PTR(ptr) \
-    __extension__ ({volatile VALUE *rb_gc_guarded_ptr = (ptr); rb_gc_guarded_ptr;})
-#else
-#ifdef _MSC_VER
+#define RB_GC_GUARD(v) \
+    (*__extension__ ({ \
+	volatile VALUE *rb_gc_guarded_ptr = &(v); \
+	__asm__("" : : "m"(rb_gc_guarded_ptr)); \
+	rb_gc_guarded_ptr; \
+    }))
```

Comment:
We probably don't want this definition of RB_GC_GUARD.
Maybe we just `#undef RB_GC_GUARD` below and redefine it as identity like before?

Reply:
Will push a fix.